src/frame/mod.rs

```diff
@@ -12,7 +12,80 @@ use std::{borrow::Cow, error, fmt};
 use crate::bytes::Bytes;
 
 /// A Modbus function code is represented by an unsigned 8 bit integer.
-pub type FunctionCode = u8;
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum FunctionCode {
+    ReadCoils,                  //=> 0x01,
+    ReadDiscreteInputs,         // => 0x02,
+    WriteSingleCoil,            //(_, _) => 0x05,
+    WriteMultipleCoils,         //(_, _) => 0x0F,
+    ReadInputRegisters,         //(_) => 0x04,
+    ReadHoldingRegisters,       //(_) => 0x03,
+    WriteSingleRegister,        //(_, _) => 0x06,
+    WriteMultipleRegisters,     //(_, _) => 0x10,
+    MaskWriteRegister,          //(_, _, _) => 0x16,
+    ReadWriteMultipleRegisters, //(_) => 0x17,
+    Custom(u8),                 // => code,
+}
+
+impl From<FunctionCode> for u8 {
+    fn from(val: FunctionCode) -> u8 {
+        match val {
+            FunctionCode::ReadCoils => 0x01,
+            FunctionCode::ReadDiscreteInputs => 0x02,
+            FunctionCode::WriteSingleCoil => 0x05,
+            FunctionCode::WriteMultipleCoils => 0x0f,
+            FunctionCode::ReadInputRegisters => 0x04,
+            FunctionCode::ReadHoldingRegisters => 0x03,
+            FunctionCode::WriteSingleRegister => 0x06,
+            FunctionCode::WriteMultipleRegisters => 0x10,
+            FunctionCode::MaskWriteRegister => 0x16,
+            FunctionCode::ReadWriteMultipleRegisters => 0x17,
+            FunctionCode::Custom(code) => code,
+        }
+    }
+}
+
+impl From<u8> for FunctionCode {
+    fn from(value: u8) -> Self {
+        match value {
+            0x01 => FunctionCode::ReadCoils,
+            0x02 => FunctionCode::ReadDiscreteInputs,
+            0x05 => FunctionCode::WriteSingleCoil,
+            0x0f => FunctionCode::WriteMultipleCoils,
+            0x04 => FunctionCode::ReadInputRegisters,
+            0x03 => FunctionCode::ReadHoldingRegisters,
+            0x06 => FunctionCode::WriteSingleRegister,
+            0x10 => FunctionCode::WriteMultipleRegisters,
+            0x16 => FunctionCode::MaskWriteRegister,
+            0x17 => FunctionCode::ReadWriteMultipleRegisters,
+            value => FunctionCode::Custom(value),
+        }
+    }
+}
+
+impl std::fmt::Display for FunctionCode {
```

Comment:
I don't think that this is a suitable implementation of std::fmt::Display. Its more a like an extended debug formatting and is not needed. Function codes should be displayed as their inner `u8` type.

Reply:
since the whole type is removed. It's display method also doesnt exist anymore